build.assets/tooling/cmd/build-apt-repos/utils.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime/debug"
+	"strings"
+)
+
+func GetSubdirectories(basePath string) <-chan string {
```

Comment:
Is the idea here to return the `chan string` as a sort of lazy list of the subdirs? How many subdirs are you expecting?

If it's anything less than tens of thousands, I'd just do it synchronously and return `([]string, error)` - which also has the nice property of making a failure in `GetSubdirectories()` not necessarily be a showstopper.

Reply:
Fixed per your recommendations